File: app/src/main/java/com/kidozh/discuzhub/results/MessageResult.java
package com.kidozh.discuzhub.results;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.kidozh.discuzhub.entities.ErrorMessage;

import java.io.Serializable;

public class MessageResult implements Serializable {
    @JsonProperty("messagestr")
    public String content="";
    @JsonProperty("messageval")
    public String key="";

    public ErrorMessage toErrorMessage(){
        return new ErrorMessage(key,content);
    }
}

File: app/src/main/java/com/kidozh/discuzhub/results/MessageResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonProperty
import com.kidozh.discuzhub.entities.ErrorMessage
import java.io.Serializable

class MessageResult : Serializable {
    @JsonProperty("messagestr")
    var content = ""

    @JsonProperty("messageval")
    var key = ""
    fun toErrorMessage(): ErrorMessage {
        return ErrorMessage(key, content)
    }
}